sarc/ldap/read_mila_ldap.py

```diff
@@ -245,26 +247,40 @@ def client_side_user_updates(LD_users_DB, LD_users_LDAP):
     DD_users_DB = dict((e["mila_email_username"], e) for e in LD_users_DB)
     DD_users_LDAP = dict((e["mila_email_username"], e) for e in LD_users_LDAP)
 
-    LD_users_to_update_or_insert = []
+    inserts = []
+    updates = []
     for meu in set(list(DD_users_DB.keys()) + list(DD_users_LDAP.keys())):
         # `meu` is short for the mila_email_username value
 
-        if meu in DD_users_DB and not meu in DD_users_LDAP:
+        user_is_in_ldap = meu in DD_users_LDAP
+        user_is_in_db = meu in DD_users_DB
+
+        if user_is_in_db and not user_is_in_ldap:
             # User is in DB but not in the LDAP.
             # Let's mark it as archived.
             entry = DD_users_DB[meu]
             entry["status"] = "archived"
+            entry.setdefault("end_date", datetime.today())
```

Comment:
C'est peut-être mieux de ne pas setter le end_date pour qu'on sache qu'on a une donnée manquante. Ça veux dire par contre que le query doit aussi faire attention au status et éviter les archived.

On pourrait aussi potentiellement avoir un cas où on a un user qui nous parviens de monmila, mais n'est pas encore dans le ldap. Ce user là devrait n'avoir accès à rien donc ce n'est pas trop critique, mais faudrait pas qu'on le mette faussement à archived. Je crois qu'à partir de maintenant, c'est monmila qui devrait avoir la priorité sur la détermination du status.

Reply:
Are we sure we are ever going to have the start and end date ?
to me it looks like it will be inferred with the LDAP.

Reply:
Ouin, le problème est qu'on fait une écriture dans la DB avant d'avoir l'info complète sur le user, alors on doit inférer à partir du LDAP seulement. L'info est présente dans MonMila. On va peut-être avoir besoin d'un minimum refactoring pour mettre ça au problème maintenant.